src/rdkafka_broker.c

```diff
@@ -3661,28 +3662,32 @@ rd_kafka_fetch_preferred_replica_handle (rd_kafka_toppar_t *rktp,
         const rd_ts_t five_seconds = 5*1000*1000;
         rd_kafka_broker_t *preferred_rkb;
         rd_kafka_t *rk = rktp->rktp_rkt->rkt_rk;
-        rd_ts_t new_intvl = rd_interval_immediate(&rktp->rktp_new_lease_intvl,
-                                                  one_minute, 0);
+        rd_ts_t new_intvl = rd_interval(&rktp->rktp_new_lease_intvl,
```

Comment:
just wanted to check you're 100% about this. AFAICT looking at it quickly again now, this will operate as required (first call will still return > 0, which is what is needed). regardless, i'd argue `rd_interval_immediate` is semantically more correct.

Reply:
Sorry, don't know why I changed that. Reverting.

Reply:
very good. i spent a long time making this work correctly / fprintf debugging it. although I couldn't see any problem with the code just reading the code, i wasn't sure i wasn't missing something...